pkg/executor/common.go

```diff
@@ -306,6 +309,78 @@ func AbortJob(ctx context.Context, c kubernetes.Interface, namespace string, job
 	}, nil
 }
 
+func ParseJobTemplate(cfg *config.Config) (template string, err error) {
+	var data, decoded []byte
+
+	f, err := os.Open(filepath.Join(cfg.TestkubeConfigDir, "job-template.yml"))
+	if err == nil {
+		data, err = io.ReadAll(f)
+		if err != nil {
+			return "", err
+		}
+		template = string(data)
+	} else if cfg.TestkubeTemplateJob != "" {
+		decoded, err = base64.StdEncoding.DecodeString(cfg.TestkubeTemplateJob)
+		if err != nil {
+			return "", err
+		}
+		template = string(decoded)
+	}
+
+	return template, nil
+}
+
+func ParseContainerTemplates(cfg *config.Config) (t Templates, err error) {
+	var decoded, data []byte
+
+	f, err := os.Open(filepath.Join(cfg.TestkubeConfigDir, "job-container-template.yml"))
+	if err == nil {
+		data, err = io.ReadAll(f)
+		if err != nil {
+			return t, err
+		}
+		t.Job = string(data)
+	} else if cfg.TestkubeContainerTemplateJob != "" {
+		decoded, err = base64.StdEncoding.DecodeString(cfg.TestkubeContainerTemplateJob)
+		if err != nil {
+			return t, err
+		}
+		t.Job = string(decoded)
+	}
+
+	f, err = os.Open(filepath.Join(cfg.TestkubeConfigDir, "job-scraper-template.yml"))
+	if err == nil {
+		data, err = io.ReadAll(f)
+		if err != nil {
+			return t, err
+		}
+		t.Scraper = string(data)
+	} else if cfg.TestkubeContainerTemplateScraper != "" {
+		decoded, err = base64.StdEncoding.DecodeString(cfg.TestkubeContainerTemplateScraper)
+		if err != nil {
+			return t, err
+		}
+		t.Scraper = string(decoded)
+	}
+
+	f, err = os.Open(filepath.Join(cfg.TestkubeConfigDir, "pvc-container-template.yml"))
+	if err == nil {
+		data, err = io.ReadAll(f)
+		if err != nil {
+			return t, err
+		}
+		t.PVC = string(data)
+	} else if cfg.TestkubeContainerTemplatePVC != "" {
+		decoded, err = base64.StdEncoding.DecodeString(cfg.TestkubeContainerTemplatePVC)
+		if err != nil {
+			return t, err
+		}
+		t.PVC = string(decoded)
+	}
+
+	return t, nil
+}
```

Comment:
Guess, you can do it similar to what I did for crd templates. They are embedded into binary, no need to have a separted dir with templates and copy it

Reply:
I don't fully know what is dynamic and what not.

Are the templates something that doesn't change often?

//go:embed can also be used to embed a file into a string

Reply:
executors.json is dynamic, other ones are not changes often. but if we want to mount config map,then we don't need embedded